src/coniql/types.py

```diff
@@ -32,24 +42,68 @@ def contains(self, value: float) -> bool:
 ]
 
 
-@dataclass
+@strawberry.type
 class ChannelDisplay:
+    # A human readable possibly multi-line description for a tooltip
     description: str
-    role: str
-    widget: Widget
+    # What access role does the Channel have
+    role: "ChannelRole"
+    # Default widget to display this Channel
+    widget: Optional[Widget]
+    # If numeric, the range the put value should be within
     controlRange: Optional[Range] = None
+    # If numeric, the range the current value should be within
     displayRange: Optional[Range] = None
+    # If numeric, the range outside of which an alarm will be produced
     alarmRange: Optional[Range] = None
+    # If numeric, the range outside of which a warning will be produced
     warningRange: Optional[Range] = None
+    # If numeric, the physical units for the value field
     units: Optional[str] = None
+    # If numeric, the number of decimal places to display
     precision: Optional[int] = None
+    # If numeric, how should value be displayed
     form: Optional[DisplayForm] = None
+    # If given, the value should be one of these choices
     choices: Optional[List[str]] = None
 
 
-def make_number_format_string(
-    form: Optional[DisplayForm], precision: Optional[int]
-) -> str:
+@strawberry.enum
+@dataclass
+class NumberType(Enum):
+    INT8 = "INT8"
+    UINT8 = "UINT8"
+    INT16 = "INT16"
+    UINT16 = "UINT16"
+    INT32 = "INT32"
+    UINT32 = "UINT32"
+    INT64 = "INT64"
+    UINT64 = "UINT64"
+    FLOAT32 = "FLOAT32"
+    FLOAT64 = "FLOAT64"
+
+
+@strawberry.enum
+class ChannelRole(Enum):
+    """
+    What access role has the Channel
+    """
+
+    RO = "RO"
+    WO = "WO"
+    RW = "RW"
+
+
+@strawberry.type
+@dataclass
```

Comment:
Is there a reason this class still has the `@dataclass` annotation, but `ChannelDisplay`, above, does not?